File: 553-optimal-division/optimal-division.dart
class Solution {
  String optimalDivision(List<int> nums) {
    String a = nums[0].toString();
    String b = '';
    for(int i = 1; i < nums.length; i++){
        if(i == 1 && nums.length > 2){
            b += '/(' + nums[i].toString();
        }
        else{
            b += '/' + nums[i].toString();
        }
        if(i == nums.length-1 && nums.length > 2){
            b += ')';
        }
    }
    return a+b;
  }
}

File: 553-optimal-division/optimal-division.js
/**
 * @param {number[]} nums
 * @return {string}
 */
const optimalDivision = (n) => {
    let len = n.length;
    let ans = "";
    ans += n[0];
    if(len === 0) return "";
    if(len === 1) return ans;
    if(len > 2){
        ans += '/(';
    }else{
        ans += '/';
    }

    for(let i=1;i<len;i++){
        ans += n[i];
        if( i !== len - 1) ans += '/';
    }
    
    if(len > 2) ans += ')';

    return ans;
};